src/com/amazon/ion/impl/lite/IonValueLite.java

```diff
@@ -614,7 +614,75 @@ public final SymbolToken getKnownFieldNameSymbol()
         return token;
     }
 
-    final boolean clearSymbolIDValues()
+    private static class ClearSymbolIDsHolder
+    {
+        boolean allSIDsClear = false;
+        IonContainerLite parent = null;
+        IonContainerLite.SequenceContentIterator iterator = null;
+    }
+
+    private static ClearSymbolIDsHolder[] growClearSymbolIDsHolderStack(
+            ClearSymbolIDsHolder[] clearSymbolIDsHolderStack)
+    {
+        ClearSymbolIDsHolder[] newClearSymbolIDsHolderStack = new ClearSymbolIDsHolder[clearSymbolIDsHolderStack.length * 2];
+        System.arraycopy(clearSymbolIDsHolderStack, 0, newClearSymbolIDsHolderStack, 0,
+                clearSymbolIDsHolderStack.length);
+        return newClearSymbolIDsHolderStack;
+    }
+
+    private boolean clearSymbolIDsIterative(boolean readOnlyMode) {
+        ClearSymbolIDsHolder[] clearSymbolIDsHolderStack = new ClearSymbolIDsHolder[CONTAINER_STACK_INITIAL_CAPACITY];
+        int clearSymbolIDsHolderStackIndex = -1;
+        IonValueLite value = this;
+        ClearSymbolIDsHolder currentClearSymbolIDsHolder = null;
+        boolean allSIDsClear = true;
+
+        do {
+            if (!(value instanceof IonContainerLite)) {
+                if (currentClearSymbolIDsHolder == null) {
+                    allSIDsClear = value.scalarClearSymbolIDValues();
+                } else {
+                    currentClearSymbolIDsHolder.allSIDsClear &= value.scalarClearSymbolIDValues();
+                }
+                if(readOnlyMode) {
+                    value._isLocked(true);
+                }
+            } else {
+                if (++clearSymbolIDsHolderStackIndex >= clearSymbolIDsHolderStack.length) {
+                    clearSymbolIDsHolderStack = growClearSymbolIDsHolderStack(clearSymbolIDsHolderStack);
+                }
+                currentClearSymbolIDsHolder = new ClearSymbolIDsHolder();
+                currentClearSymbolIDsHolder.parent = (IonContainerLite) value;
+                currentClearSymbolIDsHolder.iterator = currentClearSymbolIDsHolder.parent.new SequenceContentIterator(
+                        0,
+                        true);
+                currentClearSymbolIDsHolder.allSIDsClear = currentClearSymbolIDsHolder.parent.attemptClearSymbolIDValues();
```

Comment:
Why is this not `scalarClearSymbolIDValues` (with a different name, since it seems to apply to both scalars and containers)?